### ISSUE TITLE
[$250]  Dupe detect - Review duplicates button is present for employee after admin has resolved dupes

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#54167]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.76-6
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** Yes, reproducible on both
**If this was caught during regression testing, add the test name, ID and link from TestRail:** Exp
**Email or phone of affected tester (no customers):** applausetester+kh04120014@applause.expensifail.com
**Issue reported by:** Applause Internal Team
**Device used:** Mac 15.0 / Chrome
**App Component:** Money Requests

## Action Performed:
Precondition:
- Admin and employee are in the same workspace.
- Admin is approver.

1. Go to staging.new.expensify.com
2. [Employee] Submit an expense to the workspace chat.
3. [Admin] Pay the expense.
4. [Employee] Submit another same expense as Step 2 to the workspace chat.
5. [Admin] Click on the expense preview to open expense report.
6. [Admin] Click Review duplicates.
7. [Admin] Click Keep this one.
8. [Employee] Refresh the page.
9. [Employee] Go to expense report and click Review duplicates.
10. [Employee] Click Keep this one.

## Expected Result:
After admin has resolved the duplicates, Review duplicates button should disappear for employee.

## Actual Result:
In Step 9, after admin has resolved the duplicates, Review duplicates button is still present for employee.
When employee tries to resolve the duplicates after admin has resolved it, Review duplicates button reappears.

## Workaround:
Unknown

## Platforms:
- [x] Android: Standalone
- [x] Android: HybridApp
- [x] Android: mWeb Chrome
- [x] iOS: Standalone
- [x] iOS: HybridApp
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/3f47a251-7d7f-46bc-b04c-81f3b8170819

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021879237687605476132</li>
        <li>Upwork Job ID: 1879237687605476132</li>
        <li>Last Price Increase: 2025-01-21</li>
    </ul>
</details>

**Comment by zanyrenney** [comment]
Looking now.

**Comment by OlimpiaZurek** [comment]
Hi. I'm Olimpia from Callstack and I'd like to work on this issue.

**Comment by OlimpiaZurek** [comment]
Here’s my update and findings.

The `Review duplicates`  button is displayed when there are transaction violations:
https://github.com/Expensify/App/blob/264728f1b086117b9cabfd4728ff30b8afdfc520/src/libs/TransactionUtils/index.ts#L910

The  `ResolveDuplicates` command has replaced the  `MergeCollection` method:

https://github.com/Expensify/App/blob/264728f1b086117b9cabfd4728ff30b8afdfc520/src/pages/TransactionDuplicate/Confirmation.tsx#L138

When an admin clicks the button, the `ResolveDuplicates` command updates one transaction and puts the rest on hold.

The issue is that after the API call, the function doesn’t properly update `transactionViolations` with the correct data. In fact, there’s no `transactionViolation` data at all in the API response:

<img width="1065" alt="Image" src="https://github.com/user-attachments/assets/7ba1427b-6207-49c1-ba76-6ac1afb3f6bb" />
 
Even though optimistic updates are applied, the missing `transactionViolations` in the API response means the Onyx store doesn’t get updated correctly.

 As a result, the `Review duplicates` button doesn’t appear for the admin and still remains visible to employees because the `duplicatedTransaction` violation isn’t  updated correctly.

I’m not sure if there’s anything we can do from the app perspective. Ideally, the API should return the correct data for `transactionViolations`.

**Comment by zanyrenney** [comment]
aassigned @OlimpiaZurek

**Comment by zanyrenney** [comment]
Thank you!

**Comment by dylanexpensify** [comment]
Is this still occurring? Cc @zanyrenney

**Comment by zanyrenney** [comment]
Ooh sorry! Let me add the internal label for this!

**Comment by zanyrenney** [comment]
Yep! We're waiting on an internal engineer for capacity on this one. So would probs be a good one to highlight in the update @dylanexpensify ! Thanks!

**Comment by zanyrenney** [comment]
It actually shouldn't be started, it's on all of us to update. Updating it now!

**Comment by zanyrenney** [comment]
this is not overdue ^ just added a comment above :)

**Comment by zanyrenney** [comment]
@dylanexpensify mind bumping this in the daily update, please?

**Comment by zanyrenney** [comment]
bumped in expense.

**Comment by zanyrenney** [comment]
This is the expected behaviour - confirmed by @pecanoro
